src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let arr = [];
+  let cloneState = {...state};
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(cloneState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const rem of action.keysToRemove) {
+          delete cloneState[rem];
+        }
+        break;
+
+      case 'clear':
+        cloneState = {};
+        break;
+
+      default:
+        return arr;
```

Comment:
It is better to throw error in this case
```suggestion
        throw new Error("Unsupported action type");
```